### PR TITLE
Add pezhore preonic keymap

### PR DESCRIPTION
**Pull request by pezhore**
Adding my keymap to for the Preonic Rev3. This strips out the alternative keyboard layouts aside from QWERTY,
adds a NumPad layer, and an Fn layer for emoji/kamoji keys.

## Description

Adding yet another keymap for Preonic - this one has working shrug ( `¯\_(ツ)_/¯`), Table flip (`(╯°□°)╯︵ ┻━┻`), Disappointed (`(ಠ_ಠ)`), and Strut (`ᕕ( ᐛ )ᕗ`) kamoji, as well as Chocolate pudding (💩) emoji.

Colemak and Dvorak layers have been removed, and a default Numpad layer has been added.

Fn momentary layer adds:

* Control RGB effects/colors/etc
* Provide Kamoji/Emoji keys
* Swap between QWERTY/Numpad layers
* Set Unicode output mode

Lower Layer alterations:

* Removed ISO keys/all symbol keys in top two rows
* Removes media keys
* adds `Home`, `Pgdn`, `Pgup`, and `End` keys

Raise Layer alterations:

* Removes ISO keys/all symbol keys in top two rows
* Adds full media keys (including previous track key)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/keymaps/pezhore/pezonic-preonic_rev3.json [hunk 1]
```diff
@@ -0,0 +1,383 @@
+{
```
Comment (drashna) [inline keyboards/preonic/keymaps/pezhore/pezonic-preonic_rev3.json:1]: Shouldn't need both files here.

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_preonic_grid( \
+  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,  \
+  _______,  RESET,    DEBUG,    _______,  _______,  _______,  _______,  TERM_ON,  TERM_OFF, _______,  _______,  KC_DEL,  \
+  _______,  _______,  MU_MOD,   AU_ON,    AU_OFF,   AG_NORM,  AG_SWAP,  _______,  _______,  _______,  _______,  _______, \
+  _______,  MUV_DE,   MUV_IN,   MU_ON,    MU_OFF,   MI_ON,    MI_OFF,   _______,  _______,  _______,  _______,  _______, \
+  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______  \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/pezhore/keymap.c:168]:
```suggestion
[_QWERTY] = LAYOUT_preonic_grid(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
  FN,      KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
),

/* FN
 * ,-----------------------------------------------------------------------------------.
 * |RGBTgl|Brigt-|Brigt+|      |      |      |      |      |      |      |      | POOP |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * | RGB+ | Hue- | Hue+ |      |      |      |      |      |      |      |PrtScr|UC_MOD|
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * | RGB- | Sat- | Sat+ |      |      |      |      |      |      |      |      |UC_Wnc|
 * |------+------+------+------+------+------|------+------+------+------+------+------|
 * |      |Effct-|Effct+|      |      |      |      |      |      |      |      |UC_Lin|
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * |      |SHRUG |TFLIP |DPOINT|STRUT |      |      |      |QWERTY|NUMPAD|      |UC_Mac|
 * `-----------------------------------------------------------------------------------'
 */
[_FN] = LAYOUT_preonic_grid(
  RGB_TOG,  RGB_VAD, RGB_VAI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  POOP,
  RGB_MOD,  RGB_HUD, RGB_HUI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_PSCR,  UC_MOD,
  RGB_RMOD, RGB_SAD, RGB_SAI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  UC_M_WC,
  _______,  RGB_SPD, RGB_SPI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  UC_M_LN,
  _______,  SHRUG,   TFLIP,    DPOINT,   STRUT,    _______,  _______,  _______,   QWERTY,  NUMPAD,   _______,  UC_M_MA
),

/* Numpad
 * ,-----------------------------------------------------------------------------------.
 * |      |      |      |      |      |      |NumLk |   7  |   8  |   9  |   /  | Bksp |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * | Tab  |      |      |      |      |      |      |   4  |   5  |   6  |   *  | Del  |
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * | Esc  |      |      |      |      |      |      |   1  |   2  |   3  |   -  |  /   |
 * |------+------+------+------+------+------|------+------+------+------+------+------|
 * |      |      |      |      |      |      |      |   0  |   .  |   =  |   +  |Enter |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * |      |      |      |      |      |      |      |      |      |      |      |      |
 * `-----------------------------------------------------------------------------------'
 */
[_NUMPAD] = LAYOUT_preonic_grid(
  _______,  _______,  _______,  _______,  _______,  _______,  KC_NLCK,   KC_P7,   KC_P8,    KC_P9,    KC_PSLS,  KC_BSPC,
  KC_ESC,   _______,  _______,  _______,  _______,  _______,  _______,   KC_P4,   KC_P5,    KC_P6,    KC_PAST,  KC_DEL,
  KC_TAB,   _______,  _______,  _______,  _______,  _______,  _______,   KC_P1,   KC_P2,    KC_P3,    KC_PMNS,  _______,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,   KC_P0,   KC_PDOT,  KC_PEQL,  KC_PPLS,  _______,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______
),

/* Lower
 * ,-----------------------------------------------------------------------------------.
 * |      |      |      |      |      |      |      |      |      |      |      | Bksp |
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * |      |      |      |      |      |      |      |      |      |      |      | Del  |
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   _  |   +  |   {  |   }  |  |   |
 * |------+------+------+------+------+------|------+------+------+------+------+------|
 * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |      |      |      |      |      |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * |      |      |      |      |      |      |      |Raise | Home | PgDn | PgUp | End  |
 * `-----------------------------------------------------------------------------------'
 */
[_LOWER] = LAYOUT_preonic_grid(
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_BSPC,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_DEL,
  KC_DEL,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,   KC_F6,     KC_UNDS,  KC_PLUS,  KC_LCBR,  KC_RCBR,  KC_PIPE,
  _______,  KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,  KC_F12,    _______,  _______,  _______,  _______,  _______,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_HOME,  KC_PGDN,  KC_PGUP,  KC_END
),

/* Raise
 * ,-----------------------------------------------------------------------------------.
 * |      |      |      |      |      |      |      |      |      |      |      | Bksp |
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * |      |      |      |      |      |      |      |      |      |      |      | Del  |
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   -  |   =  |   [  |   ]  |  \   |
 * |------+------+------+------+------+------|------+------+------+------+------+------|
 * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |      | Prev |      |      |      |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * |      |      |      |      |Lower |      |      |      | Next | Vol- | Vol+ | Play |
 * `-----------------------------------------------------------------------------------'
 */
[_RAISE] = LAYOUT_preonic_grid(
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_BSPC,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_DEL,
  KC_DEL,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_MINS,  KC_EQL,   KC_LBRC,  KC_RBRC,  KC_BSLS,
  _______,  KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   _______,  KC_MPRV,  _______,  _______,  _______,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_MNXT,  KC_VOLD,  KC_VOLU,  KC_MPLY
),


/* Adjust (Lower + Raise)
 * ,-----------------------------------------------------------------------------------.
 * |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * |      | Reset|      |      |      |      |      |      |      |      |      |  Del |
 * |------+------+------+------+------+-------------+------+------+------+------+------|
 * |      |      |      |Aud on|AudOff|AGnorm|AGswap|      |      |      |      |      |
 * |------+------+------+------+------+------|------+------+------+------+------+------|
 * |      |Voice-|Voice+|Mus on|MusOff|MidiOn|MidOff|      |      |      |      |      |
 * |------+------+------+------+------+------+------+------+------+------+------+------|
 * |      |      |      |      |      |             |      |      |      |      |      |
 * `-----------------------------------------------------------------------------------'
 */
[_ADJUST] = LAYOUT_preonic_grid(
  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,
  _______,  RESET,    DEBUG,    _______,  _______,  _______,  _______,  TERM_ON,  TERM_OFF, _______,  _______,  KC_DEL,
  _______,  _______,  MU_MOD,   AU_ON,    AU_OFF,   AG_NORM,  AG_SWAP,  _______,  _______,  _______,  _______,  _______,
  _______,  MUV_DE,   MUV_IN,   MU_ON,    MU_OFF,   MI_ON,    MI_OFF,   _______,  _______,  _______,  _______,  _______,
  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/pezhore/keymap.c (hunk 2 above)
Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/keymap.c:168]: @fauxpark - just so I'm clear (Github is doing some fun code wrapping for this suggested fix), are you saying to drop all the `\` from the end of each line?

Comment (fauxpark) [reply keyboards/preonic/keymaps/pezhore/keymap.c:168]: Yes, they're only needed when actually `#define`ing the layout macros, to tell the preprocessor that the macro continues on the next line.

> do you mind if I also edit the default keymap (which has those backslashes)?

Go right ahead.

File: keyboards/preonic/keymaps/pezhore/pezonic-preonic_rev3.json (hunk 1 above)
Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/pezonic-preonic_rev3.json]: Removed in my latest commit!

File: keyboards/preonic/keymaps/pezhore/keymap.c (hunk 2 above)
Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/keymap.c]: Fixed in the latest commit!

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      register_code(KC_PGDN);
+      unregister_code(KC_PGDN);
+    } else {
+      register_code(KC_PGUP);
+      unregister_code(KC_PGUP);
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/pezhore/keymap.c:277]:
```suggestion
      tap_code(KC_PGDN);
    } else {
      tap_code(KC_PGUP);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  LOWER,
+  RAISE,
+  SHRUG,  // ¯\_(ツ)_/¯
+  TFLIP,  // (╯°□°)╯︵ ┻━┻
+  POOP,   // 💩
+  DPOINT, // (ಠ_ಠ)
+  STRUT   // ᕕ( ᐛ )ᕗ
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (noroadsleft) [inline keyboards/preonic/keymaps/pezhore/keymap.c:42]: Just to keep the keymap aligned nicely.

```suggestion
#define TG_NUMP TG(_NUMPAD)

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |SHRUG |TFLIP |DPOINT|STRUT |      |      |      |QWERTY|NUMPAD|      |UC_Mac|
+ * `-----------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_preonic_grid(
+  RGB_TOG,  RGB_VAD, RGB_VAI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  POOP,
+  RGB_MOD,  RGB_HUD, RGB_HUI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_PSCR,  UC_MOD,
+  RGB_RMOD, RGB_SAD, RGB_SAI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  UC_M_WC,
+  _______,  RGB_SPD, RGB_SPI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  UC_M_LN,
+  _______,  SHRUG,   TFLIP,    DPOINT,   STRUT,    _______,  _______,  _______,   QWERTY,  NUMPAD,   _______,  UC_M_MA
```
Comment (noroadsleft) [inline keyboards/preonic/keymaps/pezhore/keymap.c:83]: Also frees up the key that was `QWERTY` for something else, if you want.

```suggestion
  _______,  SHRUG,   TFLIP,    DPOINT,   STRUT,    _______,  _______,  _______,   _______, TG_NUMP,   _______,  UC_M_MA
```

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+ * ,-----------------------------------------------------------------------------------.
+ * |RGBTgl|Brigt-|Brigt+|      |      |      |      |      |      |      |      | POOP |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | RGB+ | Hue- | Hue+ |      |      |      |      |      |      |      |PrtScr|UC_MOD|
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | RGB- | Sat- | Sat+ |      |      |      |      |      |      |      |      |UC_Wnc|
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |Effct-|Effct+|      |      |      |      |      |      |      |      |UC_Lin|
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |SHRUG |TFLIP |DPOINT|STRUT |      |      |      |QWERTY|NUMPAD|      |UC_Mac|
```
Comment (noroadsleft) [inline keyboards/preonic/keymaps/pezhore/keymap.c:75]: Or however you want it phrased:

```suggestion
 * |      |SHRUG |TFLIP |DPOINT|STRUT |      |      |      |      |TgNmpd|      |UC_Mac|
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+        case QWERTY:
+          if (record->event.pressed) {
+            set_single_persistent_default_layer(_QWERTY);
+          }
+          return false;
+          break;
```
[the remark this answers is not in the document]
Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/keymap.c:181]: If I want to use the numpad for some period of time (like mucking about with spreadsheets), will the `TG(_NUMPAD)` let the numpad layer stay active until I deactivate it?

[the remark this answers is not in the document]
Comment (noroadsleft) [reply keyboards/preonic/keymaps/pezhore/keymap.c:181]: Yes. The first tap will turn the Numpad layer on. The next will turn it off.

- [Layers: Switching and Toggling Layers - QMK Firmware Docs](https://docs.qmk.fm/#/feature_layers?id=switching-and-toggling-layers)

File: keyboards/preonic/keymaps/pezhore/config.h [hunk 8]
```diff
@@ -0,0 +1,37 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/config.h:17]: added the standard one from keymap

**Review by noroadsleft — APPROVED** [review]

File: keyboards/preonic/keymaps/pezhore/keymap.c [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+        case FN:
+          if (record->event.pressed) {
+            layer_on(_FN);
+          } else {
+            layer_off(_FN);
+          }
+          return false;
+          break;
```
Comment (drashna) [inline keyboards/preonic/keymaps/pezhore/keymap.c:183]: You don't actually need this. You can just use `MO(FN)` instead.  You don't need  a custom macro for this.

Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/keymap.c:183]: @drashna - by removing those lines and changing the layout for QWERTY to something like this?

```
[_QWERTY] = LAYOUT_preonic_grid(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
  MO(FN),  KC_LCTL, KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
),
```

if so, I just tried that and it looks like that breaks the unicode keys...

Comment (drashna) [reply keyboards/preonic/keymaps/pezhore/keymap.c:183]: Well, you have `MO(FN)`, that's wrong. It should be `MO(_FN)`.  It's a subtle difference, but makes a massive difference.

**Review by fauxpark — APPROVED** [review]

File: keyboards/preonic/keymaps/pezhore/keymap.c (hunk 9 above)
Comment (pezhore) [reply keyboards/preonic/keymaps/pezhore/keymap.c:183]: You're right, it was a long day yesterday and I confused the keycode with the layer.

With that change in place, do I still need the fn line below:

```
enum preonic_keycodes {
  FN = SAFE_RANGE,
  LOWER,
  RAISE,
  SHRUG,  // ¯\_(ツ)_/¯
  TFLIP,  // (╯°□°)╯︵ ┻━┻
  POOP,   // 💩
  DPOINT, // (ಠ_ಠ)
  STRUT   // ᕕ( ᐛ )ᕗ
};
```

Comment (noroadsleft) [reply keyboards/preonic/keymaps/pezhore/keymap.c]: As you've probably already discovered, you don't need to add `FN` to the keycodes `enum`.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**